### ISSUE TITLE
SATA SSD not recogised..

### ISSUE THREAD

**Issue by hubhab** [SchmockLord/Hackintosh-Intel-i9-10900k-Gigabyte-Z490-Vision-D#140]
First thanks to Chris for the share and big groundwork done for us users..

I’ve installed Catalina with the opencore 0.6.5 template from here. my two m2 pci drives have been working without issues (after disabling sleep, otherwise they wouldn’t run at full speed, oddly enough). 

But my 2x 8TB Samsung SATA SSD drives don’t show up. Well, actually they occasionally did, but then on next few boots don’t come up anymore, very strange. 

Any idea what could have gone wrong here?

Specs:
Intel i9 10900K CPU
Gigabyte Mainboard Z490 VISION D
Gigabyte Aorus Radeon RX 5700 XT 
2x Corsair Force MP510
2x Samsung SSD 870 QVO 2.5" 8 TB 
Fenvi FV-T919 Wireless/Bluetooth Adapter (für MacOS)

**Comment by RobyRew** [comment]
You need to add the proper kext for SATA Controller.

**Comment by MirkoMela** [comment]
it's normal, Sata sharing with M.2 , impossible to use all port. Try to use another Sata port.

**Comment by hubhab** [comment]
Thanks for your replies.. 

@RobyRew I thought it was already included, and the sata drives have shown up on occasion but mostly don’t, so bit strange.. will double check to make sure.. thanks..

@MirkoMela Have taken that in consideration when building and chosen SATA 2&3 which aren‘t shared.. also I had zero issues on windows.. so sadly that’s not it..

**Comment by mauticom** [comment]
Indee Gigabyte did a shitty and misleading job when promoting the board that suggests you can use all 6 SATA ports with 3 M2s. I'm running now my win M2 from top slot(1) and macOS M2 from the bottom slot(3) so I can at least use 4 SATA ports out of the 6.

Also try:

- Run Samsung Magician under Windows and update to the latest SSDs Firmware. This might help.

- Also since the occasionally show up you also should double check your SATA cables and may try a different one.

**Comment by MirkoMela** [comment]
> Thanks for your replies..
> 
> @RobyRew I thought it was already included, and the sata drives have shown up on occasion but mostly don’t, so bit strange.. will double check to make sure.. thanks..
> 
> @MirkoMela Have taken that in consideration when building and chosen SATA 2&3 which aren‘t shared.. also I had zero issues on windows.. so sadly that’s not it..

Yep but remember a real Mac have only max 2 port Sata . Like 15 port max for usb

**Comment by mauticom** [comment]
I have never heard of macOS having any SATA limits. I'm currently running the Vision D with Big Sur with 2 M2s and 3 SATA drives and used 5 out of 6 SATA ports in my old Mac Pro.

**Comment by hubhab** [comment]
@mauticom thanks for your feedback.. I have done the same, also used first and third row of the M2 slots.. left SATA0 and SATA1 empty and connected my SATA SSDs to SATA2 and SATA3.. will shortly get a third to connect to SATA4..

I will try upgrading the firmware after I double checked the SATA kext.. and updating to opencore 0.6.6.. haven’t had time lately to look further into this issue due too much workload.. I’ve seen other users fix issues after updating their SSD firmware so who knows..  

I don’t think it’s the cables, as it’s happening on both SATA drives and I’ve had no such problems on windows. So it’s solely an OSX issue..

**Comment by hubhab** [comment]
@MirkoMela thanks for your further input. I’m aware of the USB restrictions.. but have never heard of the SATA restrictions being so low. where’s the source of that information I’m wondering? First time I have heard that..

Nether the less, I’ve only occupied two SATA channels, so don’t think that would be causing the issue either if it were so..

**Comment by hubhab** [comment]
for those who run into same issues as I did on this MoBo.. seems some shared SATA ports weren't disabled in BIOS and caused my issue in hackintosh.. even though I had my SATA drives on non shared SATA ports.. weird, but it did the trick..
